File: tournaments/arrayChange/arrayChange.cpp
int arrayChange(std::vector<int> inputArray) {

  int result = 0;

  for (int i = 1; i < inputArray.size(); i++) {
    if (inputArray[i] <= inputArray[i - 1]) {
      result += inputArray[i - 1] - inputArray[i] + 1;
      inputArray[i] = inputArray[i - 1] + 1;
    }
  }
  return result;
}

File: tournaments/arrayChange/arrayChange.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub arrayChange {
    my ($inputArray) = @_;

    my $result = 0;

    for (my $i = 1; @$inputArray > $i; ++$i) {
        if (@$inputArray[$i] <= @$inputArray[$i - 1]) {
            $result += (1 + @$inputArray[$i - 1] - @$inputArray[$i]);
            @$inputArray[$i] = 1 + @$inputArray[$i - 1];
        }
    }

    return $result;
}
